beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -546,3 +567,40 @@ func (f *ForkChoice) SetGenesisTime(genesisTime uint64) {
 func (f *ForkChoice) SetOriginRoot(root [32]byte) {
 	f.store.originRoot = root
 }
+
+// CachedHeadRoot returns the last cached head root
```

Comment:
This method is a fast access to the last  computed head. A future possible implementation would be to actually compute head when inserting a node. when processing attestations and on `on_tick`. Then a call to `Head()` would just return the cached value.